packages/flutter/lib/src/rendering/sliver_app_bar.dart

```diff
@@ -0,0 +1,308 @@
+  double _lastShrinkOffset;
+
+  /// Called during layout if the shrink offset has changed.
+  ///
+  /// During this callback, the [child] can be set, mutated, or replaced.
+  @protected
+  void updateChild(double shrinkOffset) { }
+
+  /// Flag the current child as stale and needing updating even if the shrink
+  /// offset has not changed.
+  ///
+  /// Call this whenever [updateChild] would change or mutate the child even if
+  /// given the same `shrinkOffset` as the last time it was called.
+  ///
+  /// This must be implemented by [RenderSliverAppBar] subclasses such that the
+  /// next layout after this call will result in [updateChild] being called.
+  @protected
+  void markNeedsUpdate() {
+    markNeedsLayout();
+    _lastShrinkOffset = null;
+  }
+
+  void layoutChild(double scrollOffset, double maxExtent) {
+    assert(maxExtent != null);
+    double shrinkOffset = math.min(scrollOffset, maxExtent);
+    if (shrinkOffset != _lastShrinkOffset) {
+      invokeLayoutCallback/*<SliverConstraints>*/((SliverConstraints constraints) {
+        assert(constraints == this.constraints);
+        updateChild(shrinkOffset);
+        _minExtent = _getChildIntrinsicExtent();
+      });
+      _lastShrinkOffset = shrinkOffset;
+    }
+    assert(_minExtent != null);
+    assert(() {
+      if (_minExtent <= maxExtent)
+        return true;
+      throw new FlutterError(
+        'The maxExtent for this $runtimeType is less than the child\'s intrinsic extent.\n'
+        'The specified maxExtent was: ${maxExtent.toStringAsFixed(1)}\n'
+        'The child was updated with shrink offset: ${shrinkOffset.toStringAsFixed(1)}\n'
+        'The actual measured intrinsic extent of the child was: ${_minExtent.toStringAsFixed(1)}\n'
+      );
+    });
+    child?.layout(
+      constraints.asBoxConstraints(maxExtent: math.max(_minExtent, maxExtent - shrinkOffset)),
+      parentUsesSize: true,
+    );
+  }
+
+  /// Returns the distance from the leading _visible_ edge of the sliver to the
+  /// side of the child closest to that edge, in the scroll axis direction.
+  ///
+  /// For example, if the [constraints] describe this sliver as having an axis
+  /// direction of [AxisDirection.down], then this is the distance from the top
+  /// of the visible portion of the sliver to the top of the child. If the child
+  /// is scrolled partially off the top of the viewport, then this will be
+  /// negative. On the other hand, if the [constraints] describe this sliver as
+  /// having an axis direction of [AxisDirection.up], then this is the distance
+  /// from the bottom of the visible portion of the sliver to the bottom of the
+  /// child. In both cases, this is the direction of increasing
+  /// [SliverConstraints.scrollOffset].
+  ///
+  /// Calling this when the child is not visible is not valid.
+  ///
+  /// The argument must be the value of the [child] property.
+  ///
+  /// This must be implemented by [RenderSliverAppBar] subclasses.
```

Comment:
I changed the implementation to call the superclass implementation which throws.

This implementation is here just to provide more precise documentation.